mbed_host_tests/host_tests_runner/mbed_base.py

```diff
@@ -70,19 +75,35 @@ def __init__(self, options):
             # We need to normalize path before we open file
             json_test_configuration_path = self.options.json_test_configuration.strip("\"'")
             try:
-                print "MBED: Loading test configuration from '%s'..." % json_test_configuration_path
+                self.logger.prn_inf("Loading test configuration from '%s'..." % json_test_configuration_path)
                 with open(json_test_configuration_path) as data_file:
                     self.test_cfg = json.load(data_file)
             except IOError as e:
-                print "MBED: Test configuration JSON file '{0}' I/O error({1}): {2}".format(json_test_configuration_path, e.errno, e.strerror)
+                self.logger.prn_err("Test configuration JSON file '{0}' I/O error({1}): {2}"
+                                    .format(json_test_configuration_path, e.errno, e.strerror))
             except:
-                print "MBED: Test configuration JSON Unexpected error:", str(e)
+                self.logger.prn_err("Test configuration JSON Unexpected error:", str(e))
                 raise
 
-    def copy_image(self, image_path=None, disk=None, copy_method=None, port=None):
+    def copy_image(self, image_path=None, disk=None, copy_method=None, port=None, retry_copy=3):
         """! Closure for copy_image_raw() method.
         @return Returns result from copy plugin
         """
+
+        def get_remount_count(disk_path, tries=2):
+            for cur_try in range(1, tries + 1):
+                try:
+                    files_on_disk = [x.upper() for x in os.listdir(disk_path)]
+                    if 'DETAILS.TXT' in files_on_disk:
+                        with open(os.path.join(disk_path, 'DETAILS.TXT'), 'r') as details_txt:
+                            for line in details_txt.readlines():
+                                if 'Remount count:' in line:
+                                    return int(line.replace('Remount count: ', ''))
+                except OSError as e:
+                    self.logger.prn_err("Failed to get remount count due to OSError.", str(e))
+                    self.logger.prn_inf("Retrying in 1 second (try %s of %s)" % (cur_try, tries))
+                    sleep(1)
```

Comment:
@mazimkhan, after some offline discussion with @deepikabhavnani, we determined that correct behavior should be that if no `DETAILS.TXT` is detected OR if `DETAILS.TXT` is present but the `Remount count` line isn't present, then the function should return immediately. This is because both conditions indicate the firmware present on the board doesn't support the "Remount count" feature. If it isn't possible to determine these conditions due to an `OSError`, then we sleep for a bit to give the OS time to properly enumerate the device before trying again.